qiskit/providers/basic_provider/basic_simulator.py

```diff
@@ -555,55 +458,52 @@ def run(
                 )
             else:
                 out_options[key] = value
-        qobj = _assemble(run_input, self, **out_options)
-        qobj_options = qobj.config
-        self._set_options(qobj_config=qobj_options, backend_options=backend_options)
+        self._set_options(backend_options=backend_options)
         job_id = str(uuid.uuid4())
-        job = BasicProviderJob(self, job_id, self._run_job(job_id, qobj))
+        job = BasicProviderJob(self, job_id, self._run_job(job_id, run_input))
         return job
 
-    def _run_job(self, job_id: str, qobj: QasmQobj) -> Result:
-        """Run experiments in qobj
+    def _run_job(self, job_id: str, run_input) -> Result:
+        """Run circuits in job.
 
         Args:
             job_id: unique id for the job.
-            qobj: job description
+            run_input: circuits to be run.
 
         Returns:
             Result object
         """
-        self._validate(qobj)
+        if isinstance(run_input, QuantumCircuit):
+            run_input = [run_input]
+
+        self._validate(run_input)
         result_list = []
-        self._shots = qobj.config.shots
-        self._memory = getattr(qobj.config, "memory", False)
-        self._qobj_config = qobj.config
+        self._memory = True
         start = time.time()
-        for experiment in qobj.experiments:
-            result_list.append(self.run_experiment(experiment))
+        for circuit in run_input:
+            result_list.append(self._run_circuit(circuit))
         end = time.time()
         result = {
             "backend_name": self.name,
             "backend_version": self.backend_version,
-            "qobj_id": qobj.qobj_id,
+            "qobj_id": None,
             "job_id": job_id,
             "results": result_list,
             "status": "COMPLETED",
             "success": True,
             "time_taken": (end - start),
-            "header": qobj.header.to_dict(),
         }
 
         return Result.from_dict(result)
 
-    def run_experiment(self, experiment: QasmQobjExperiment) -> dict[str, ...]:
-        """Run an experiment (circuit) and return a single experiment result.
+    def _run_circuit(self, circuit) -> dict:
+        """Simulate a single circuit run.
```

Comment:
In the Returns section below, it would be nice to mention the `header` field which was added

Reply:
Added in [e14ae50](https://github.com/Qiskit/qiskit/pull/13743/commits/e14ae5058884646e3275130c2b3addd8f942f080)